### PR TITLE
added sorted & unsorted ziz-zag array and modified related readme.md

### PR DESCRIPTION
**Pull request by imsushant12**
I have added the sorted and unsorted zig-zag array in cpp and updated corresponding readme.md file

ASSIGNED ISSUE : 
link : https://github.com/Algo-Phantoms/Algo-Tree/issues/234

### CODE REVIEW

**Comment by plazzy99** [comment]
Resolve conflicts

**Comment by imsushant12** [comment]
@raksha009 and @OjusWiZard please review.

File: Code/C++/sorted_zig_zag_array.cpp [hunk 1]
```diff
@@ -0,0 +1,53 @@
+#include <bits/stdc++.h>
```
Comment (rish-singhal) [inline Code/C++/sorted_zig_zag_array.cpp:1]: add description at the top

File: Code/C++/sorted_zig_zag_array.cpp [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+        cout<<a[i]<<"  ";
+}
+
+int main()
+{
+    int a[] = {4,3,7,8,6,2,1};
+    int n = sizeof(a)/sizeof(a[0]);
+    sort(a , a+n);
+
+    cout<<"Original array  : ";
```
Comment (rish-singhal) [inline Code/C++/sorted_zig_zag_array.cpp:30]: add spaces around operators everywhere "<<, ==, ...."

File: Code/C++/unsorted_zig_zag_array.cpp [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+    zigzag(a, n);
+
+    return 0;
+}
+
+/*
+    Test Case -
+
+    Output :
+
```
Comment (rish-singhal) [inline Code/C++/unsorted_zig_zag_array.cpp:41]: atleast add 2 test_cases

File: Code/C++/unsorted_zig_zag_array.cpp [hunk 4]
```diff
@@ -0,0 +1,50 @@
+#include <bits/stdc++.h>
```
Comment (rish-singhal) [inline Code/C++/unsorted_zig_zag_array.cpp:1]: same changes required as above.

**Review by rish-singhal — CHANGES_REQUESTED** [review]

File: Code/C++/sorted_zig_zag_array.cpp (hunk 1 above)
Comment (imsushant12) [reply Code/C++/sorted_zig_zag_array.cpp]: done sir, please review

File: Code/C++/unsorted_zig_zag_array.cpp (hunk 3 above)
Comment (imsushant12) [reply Code/C++/unsorted_zig_zag_array.cpp]: done sir, please review

File: Code/C++/sorted_zig_zag_array.cpp (hunk 2 above)
Comment (imsushant12) [reply Code/C++/sorted_zig_zag_array.cpp]: done sir, please review

File: Code/C++/unsorted_zig_zag_array.cpp (hunk 4 above)
Comment (imsushant12) [reply Code/C++/unsorted_zig_zag_array.cpp]: done sir, please review

File: Array/readme.md [hunk 5]
```diff
@@ -38,10 +38,24 @@ To create an array, you need to known the length (or size) of the array in advan
...
  * Search in sorted rotated array ----> [Python](/Code/Python/search_in_sorted_rotated_array.py)
  * Sliding Window Maximum ---->[C++](/Code/C++/Sliding_Window_Maximum.cpp)
+ * Sorted Zig-Zag Array ----> [C++](/Code/C++/sorted_zig_zag_array.cpp)
  * Three Sum ----> [Python](/Code/Python/Three_Sum.py)
- * Union of two Unsorted Arrays ----> [C++](/Code/C++/Union_of_two_unsorted_array.cpp)
- * Rotation Count in a sorted array ----> [Python](/Code/Python/RotationCount.py)
+ * Union of two Arrays ----> [C++](Code/C++/Union_of_two_unsorted_array.cpp)
+ * Unsorted Zig-Zag Array ----> [C++](/Code/C++/unsorted_zig_zag_array.cpp)
+ 
+- Find duplicate in array ----> [C++](/Code/C++/Duplicate_in_array.cpp)
```
Comment (rish-singhal) [inline Array/readme.md:50]: sort this list.

File: Code/C++/sorted_zig_zag_array.cpp [hunk 6]
```diff
@@ -0,0 +1,54 @@
...
+
+    Sample Input/Output (1):
+    Original array  : 1  2  3  4  6  7  8
+    Zig - Zag array : 1  3  2  6  4  8  7
+    
+    Sample Input/Output (2):
+    Original array  : 11  12  13  14  16  17  18
+    Zig - Zag array : 11  13  12  16  14  18  17
+
+    Time Complexity : O(n log n)
```
Comment (rish-singhal) [inline Code/C++/sorted_zig_zag_array.cpp:51]: describe what n is

File: Code/C++/unsorted_zig_zag_array.cpp [hunk 7]
```diff
@@ -0,0 +1,55 @@
...
+
+    Sample Input/Output (1):
+    Original array  : 4  3  7  8  6  2  1
+    Zig - Zag array : 4  7  3  6  8  1  2
+    
+    Sample Input/Output (2):
+    Original array  : 14  13  17  18  16  12  11
+    Zig - Zag array : 14  17  13  16  18  11  12
+
+    Time Complexity : O(n)
```
Comment (rish-singhal) [inline Code/C++/unsorted_zig_zag_array.cpp:50]: describe what n is

**Review by rish-singhal — CHANGES_REQUESTED** [review]
only 3 minor changes!

**Comment by rish-singhal** [comment]
Resolve conflicts too!

File: Code/C++/unsorted_zig_zag_array.cpp (hunk 7 above)
Comment (imsushant12) [reply Code/C++/unsorted_zig_zag_array.cpp]: done sir.

File: Code/C++/sorted_zig_zag_array.cpp (hunk 6 above)
Comment (imsushant12) [reply Code/C++/sorted_zig_zag_array.cpp]: done sir

File: Array/readme.md (hunk 5 above)
Comment (imsushant12) [reply Array/readme.md:50]: done sir

**Pull request merged**